transportable-udfs-avro/src/test/java/com/linkedin/transport/avro/TestAvroWrapper.java

```diff
@@ -0,0 +1,260 @@
+/**
+ * Copyright 2018 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.transport.avro;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.linkedin.transport.api.data.PlatformData;
+import com.linkedin.transport.api.data.StdData;
+import com.linkedin.transport.api.types.StdType;
+import com.linkedin.transport.avro.data.AvroArray;
+import com.linkedin.transport.avro.data.AvroBinary;
+import com.linkedin.transport.avro.data.AvroBoolean;
+import com.linkedin.transport.avro.data.AvroDouble;
+import com.linkedin.transport.avro.data.AvroFloat;
+import com.linkedin.transport.avro.data.AvroInteger;
+import com.linkedin.transport.avro.data.AvroLong;
+import com.linkedin.transport.avro.data.AvroMap;
+import com.linkedin.transport.avro.data.AvroString;
+import com.linkedin.transport.avro.data.AvroStruct;
+import com.linkedin.transport.avro.types.AvroArrayType;
+import com.linkedin.transport.avro.types.AvroBinaryType;
+import com.linkedin.transport.avro.types.AvroBooleanType;
+import com.linkedin.transport.avro.types.AvroDoubleType;
+import com.linkedin.transport.avro.types.AvroFloatType;
+import com.linkedin.transport.avro.types.AvroIntegerType;
+import com.linkedin.transport.avro.types.AvroLongType;
+import com.linkedin.transport.avro.types.AvroMapType;
+import com.linkedin.transport.avro.types.AvroStringType;
+import com.linkedin.transport.avro.types.AvroStructType;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Map;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericArray;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.util.Utf8;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.*;
+
+
+public class TestAvroWrapper {
+
+  private Schema createSchema(String typeName) {
+    return createSchema("testField", typeName);
+  }
+
+  private Schema createSchema(String fieldName, String typeName) {
+    return new Schema.Parser().parse(
+        String.format("{\"name\": \"%s\",\"type\": %s}", fieldName, typeName));
+  }
+
+  @Test
+  public void testBooleanType() {
+    Schema boolSchema = createSchema("\"boolean\"");
+
+    StdType stdBoolType = AvroWrapper.createStdType(boolSchema);
+    assertTrue(stdBoolType instanceof AvroBooleanType);
+    assertEquals(boolSchema, stdBoolType.underlyingType());
+
+    StdData stdBoolData = AvroWrapper.createStdData(true, boolSchema);
+    assertTrue(stdBoolData instanceof AvroBoolean);
+    assertEquals(true, ((AvroBoolean) stdBoolData).get());
```

Comment:
This code is repeated a lot throughout all tests. Any way to dedup this?

Reply:
Good catch. Fixed by putting all common code for simple types in a shared method.